xbmc/utils/FileUtils.cpp

```diff
@@ -201,3 +203,59 @@ CDateTime CFileUtils::GetModificationDate(const std::string& strFileNameAndPath,
   }
   return dateAdded;
 }
+
+bool CFileUtils::CheckFileAccessAllowed(const std::string &filePath)
+{
+  // DENY access to paths matching
+  const std::vector<std::string> blacklist = {
+    "passwords.xml",
+    "sources.xml",
+    "guisettings.xml",
+    "advancedsettings.xml",
+    ".ssh",
+    "/etc"
+  };
+  // ALLOW kodi paths
+  // TODO: most likely many more needed
+  const std::vector<std::string> whitelist = {
+    CSpecialProtocol::TranslatePath("special://home"),
+    CSpecialProtocol::TranslatePath("special://xbmc")
+  };
+
+  std::string decodePath = CURL::Decode(filePath);
+  // check blacklist
+  for (const auto &b : blacklist)
+  {
+    if (decodePath.find(b) != std::string::npos)
+    {
+      CLog::Log(LOGERROR,"%s denied access to %s",  __FUNCTION__, decodePath.c_str());
+      return false;
+    }
+  }
+
+#if defined(TARGET_POSIX)
```

Comment:
No whitelist check for non-POSIX?

Reply:
let me quote the PR description: "Windows is not implemented yet, can someone please help?"

Reply:
Reading is an underappreciated skill. Next time just hit me please :bowing_man: 

That said, wouldn't it be enough to replace the `realpath` part on other platforms? The logic itself should stay the same, right?